vendor/bat-native-ads/src/bat/ads/internal/database/tables/creative_new_tab_page_ads_database_table.cc

```diff
@@ -0,0 +1,533 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/database/tables/creative_new_tab_page_ads_database_table.h"
+
+#include <functional>
+#include <set>
+#include <utility>
+
+#include "base/strings/string_util.h"
+#include "base/strings/stringprintf.h"
+#include "bat/ads/internal/ads_impl.h"
+#include "bat/ads/internal/container_util.h"
+#include "bat/ads/internal/database/database_statement_util.h"
+#include "bat/ads/internal/database/database_table_util.h"
+#include "bat/ads/internal/database/database_util.h"
+#include "bat/ads/internal/logging.h"
+#include "bat/ads/internal/time_util.h"
+
+namespace ads {
+namespace database {
+namespace table {
+
+using std::placeholders::_1;
+
+namespace {
+
+const char kTableName[] = "creative_new_tab_page_ads";
+
+const int kDefaultBatchSize = 50;
+
+}  // namespace
+
+CreativeNewTabPageAds::CreativeNewTabPageAds(
+    AdsImpl* ads)
+    : batch_size_(kDefaultBatchSize),
+      ads_(ads),
+      campaigns_database_table_(std::make_unique<Campaigns>(ads_)),
+      categories_database_table_(std::make_unique<Categories>(ads_)),
+      creative_ads_database_table_(std::make_unique<CreativeAds>(ads_)),
+      geo_targets_database_table_(std::make_unique<GeoTargets>(ads_)) {
+  DCHECK(ads_);
+}
+
+CreativeNewTabPageAds::~CreativeNewTabPageAds() = default;
+
+void CreativeNewTabPageAds::Save(
+    const CreativeNewTabPageAdList& creative_new_tab_page_ads,
+    ResultCallback callback) {
+  if (creative_new_tab_page_ads.empty()) {
+    callback(Result::SUCCESS);
+    return;
+  }
+
+  DBTransactionPtr transaction = DBTransaction::New();
+
+  // TODO(https://github.com/brave/brave-browser/issues/3661) instead of
+  // rebuilding the catalog database each time
+  Delete(transaction.get(), get_table_name());
```

Comment:
Perhaps use `Drop()` instead?

Reply:
As we drop the bundle database every time we update the catalog I agree we should remove the migration code and recreate the table. However, as discussed this should be in another issue/PR